src/Xamarin.Android.Build.Tasks/Tasks/CalculateLayoutCodeBehind.cs

```diff
@@ -220,6 +220,13 @@ bool LoadLayout (string filePath, Dictionary <string, LayoutGroup> groupIndex, r
 			bool skipFirst = false;
 
 			nav.MoveToFirstChild ();
+
+			// This is needed in case the first element after XML declaration is not an actual element but a
+			// comment, for instance
+			while (nav.NodeType != XPathNodeType.Element) {
+				nav.MoveToNext ();
```

Comment:
wow that is ugly..

So to nit pick :) what if the xml doesn't contain any elements? e.g 

```
<?xml version="1.0" encoding="utf-8"?>
 <!-- This comment is here to test that the generator
      properly skips non-element nodes preceding the
      first element node in the layout file -->
```

Reply:
Then it's going to fail quietly - it won't find any elements.

Reply:
but I'll add a test for this later today

Reply:
I added an empty file like that, but it fails in the Aapt task:

```Aapt Task (TaskId:89)
    AssetDirectory:  (TaskId:89)
    ManifestFiles:  (TaskId:89)
      /tmp/fl467ok0.jxb/AndroidManifest.xml (TaskId:89)
    ResourceDirectory: obj/Debug/res/ (TaskId:89)
    JavaDesignerOutputDirectory: /tmp/fl467ok0.jxb (TaskId:89)
    PackageName: com.xamarin.CodeBehindBuildTests (TaskId:89)
    UncompressedFileExtensions:  (TaskId:89)
    ExtraPackages:  (TaskId:89)
    AdditionalResourceDirectories:  (TaskId:89)
      /home/grendel/vc/xamarin/xamarin-android/tests/CodeBehind/BuildTests/obj/Debug/lp/0/jl/res (TaskId:89)
    AdditionalAndroidResourcePaths:  (TaskId:89)
    LibraryProjectJars:  (TaskId:89)
    ExtraArgs:  (TaskId:89)
    CreatePackagePerAbi: False (TaskId:89)
    ResourceNameCaseMap: layout/empty_layout.axml|layout/empty_layout.xml;layout/Main.axml|layout/main.xml;layout-land/Main.axml|layout-land/main.xml;values/Strings.xml|values/strings.xml;mipmap-hdpi/Icon.png|mipmap-hdpi/icon.png;mipmap-mdpi/Icon.png|mipmap-mdpi/icon.png;mipmap-xhdpi/Icon.png|mipmap
-xhdpi/icon.png;mipmap-xxhdpi/Icon.png|mipmap-xxhdpi/icon.png;mipmap-xxxhdpi/Icon.png|mipmap-xxxhdpi/icon.png; (TaskId:89)
    VersionCodePattern:  (TaskId:89)
    VersionCodeProperties:  (TaskId:89)
  Executing package -f -m -M /tmp/fl467ok0.jxb/manifest/AndroidManifest.xml -J /tmp/fl467ok0.jxb --custom-package com.xamarin.codebehindbuildtests -F /tmp/fl467ok0.jxb/resources.apk.bk -S /home/grendel/vc/xamarin/xamarin-android/tests/CodeBehind/BuildTests/obj/Debug/res/ -S /home/grendel/vc/xamarin/
xamarin-android/tests/CodeBehind/BuildTests/obj/Debug/lp/0/jl/res -I /home/grendel/android/android-sdk/platforms/android-27/android.jar --auto-add-overlay --output-text-symbols obj/Debug/ (TaskId:89)
Resources/layout/empty_layout.axml(6): warning APT0000: Error parsing XML: no element found [/home/grendel/vc/xamarin/xamarin-android/tests/CodeBehind/BuildTests/CodeBehindBuildTests.csproj]
  The previous error was converted to a warning because the task was called with ContinueOnError=true. (TaskId:89)
  Build continuing because "ContinueOnError" on the task "Aapt" is set to "True". (TaskId:89)
Done executing task "Aapt" -- FAILED. (TaskId:89)
```

I'm going to leave the change out of this PR for now, until the above issue is fixed.